packages/@uppy/drop-target/src/index.js

```diff
@@ -25,18 +27,22 @@ module.exports = class DropTarget extends BasePlugin {
     this.removeDragOverClassTimeout = null
   }
 
-  addFiles = (files) => {
-    const descriptors = files.map((file) => ({
-      source: this.id,
-      name: file.name,
-      type: file.type,
-      data: file,
-      meta: {
+  addFiles = (files, event) => {
+    const descriptors = files.map((file) => {
+      const meta = {
         // path of the file relative to the ancestor directory the user selected.
         // e.g. 'docs/Old Prague/airbnb.pdf'
         relativePath: file.relativePath || null,
-      },
-    }))
+        ...this.setMeta?.(file, event),
+      }
+      return {
+        source: this.id,
+        name: file.name,
+        type: file.type,
+        data: file,
+        meta,
+      }
+    })
```

Comment:
Any chance you could reduce the diff here by reverting to the old code, and adding `...this.setMeta?.(file, event)` to the `meta`? I don't think there's any reason to use a const here.

Reply:
Sure, I will do this.